### PR TITLE
Fixed brave's app entitlement isn't included for non-official build

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/17622

Need to include them for non-official build to test additionally added features
like vpn locally.

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-BUILD.gn.patch [hunk 1]
```diff
@@ -49,43 +49,44 @@ index a512d736c1d603680cc0e5bfb42afbd70a799f9a..0253e7271eaec572d76ce95185cb6fa5
...
        "--scm=0",
      ]
    }
 +  copy("brave_helper_plist") { deps = [ "//brave:brave_helper_plist" ] sources = [ "$root_gen_dir/brave/brave_helper_plist_tweaked.plist" ] outputs = [ "$root_gen_dir/chrome/{{source_file_part}}" ] }
  
    compile_entitlements("entitlements") {
      entitlements_templates = [ "app/app-entitlements.plist" ]
-+    if (is_official_build) { entitlements_templates += [ "app/app-entitlements-chrome.plist", "//brave/app/app-entitlements-brave.plist" ] }
++    import("//brave/app/entitlements.gni") entitlements_templates += brave_entitlements_templates
++    if (is_official_build) { entitlements_templates += [ "app/app-entitlements-chrome.plist" ] }
```
Comment (goodov) [inline patches/chrome-BUILD.gn.patch:61]: we can fill both values in ```app/entitlements.gni``` and have only a one line patch here.

Comment (simonhong) [reply patches/chrome-BUILD.gn.patch:61]: Oops! I thought `if (is_official_build) { entitlements_templates += [ "app/app-entitlements-chrome.plist" ] }` is upsteam code

Comment (simonhong) [reply patches/chrome-BUILD.gn.patch]: Fixed.

File: app/entitlements.gni [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+brave_entitlements_templates = [ "//brave/app/app-entitlements-brave.plist" ]
+
+if (is_official_build) {
+  brave_entitlements_templates += [ "app/app-entitlements-chrome.plist" ]
```
Comment (goodov) [inline app/entitlements.gni:9]: let's put a full path here? like ```//chrome/app/...```.

Comment (simonhong) [reply app/entitlements.gni]: oops again. fixed!

**Review by goodov — APPROVED** [review]

**Pull request merged**